x/ccv/consumer/keeper/keeper.go

```diff
@@ -206,6 +207,41 @@ func (k Keeper) DeletePendingChanges(ctx sdk.Context) {
 	store.Delete(types.PendingChangesKey())
 }
 
+func (k Keeper) IsPreCCV(ctx sdk.Context) bool {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.PreCCVKey())
+	if bz != nil {
+		return true
+	}
+	return false
+}
+
+func (k Keeper) SetPreCCV(ctx sdk.Context, state *types.GenesisState) {
+	initialValSet := types.GenesisState{
+		InitialValSet: state.InitialValSet,
+	}
+	bz := k.cdc.MustMarshal(&initialValSet)
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.PreCCVKey(), bz)
+}
+
+func (k Keeper) DeletePreCCV(ctx sdk.Context) {
+	store := ctx.KVStore(k.storeKey)
+	store.Delete(types.PreCCVKey())
+}
+
+func (k Keeper) GetInitialValSet(ctx sdk.Context) []tmtypes.ValidatorUpdate {
+	store := ctx.KVStore(k.storeKey)
+	initialValSet := types.GenesisState{}
+	bz := store.Get(types.PreCCVKey())
```

Comment:
Initial valset should be stored under its own key/value imo, unless I'm missing some context on why this is needed

Reply:
Initial validator set was stored on preCCV key, since both are pretty much connected each other and it's one time use.